### PR TITLE
update actions versions in test files

### PR DESCRIPTION
**Pull request by AJFSalomon**
Actions versions for `checkout` and `setup-python` were out of date in both `test.yml` and `test-and-deploy.yml`. Here's a quick fix.

### CODE REVIEW

File: {{cookiecutter.plugin_name}}/.github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
           PLATFORM: ${{ matrix.platform }}
 
       - name: Coverage
-        uses: codecov/codecov-action@v3
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/.github/workflows/test_and_deploy.yml:64]: I cannot agree for this change. codecov-action@v4 Does not support tokenless upload (with exception for PR for open source projects).

Comment (AJFSalomon) [reply {{cookiecutter.plugin_name}}/.github/workflows/test_and_deploy.yml]: I didn't know. I reversed the modification.

Comment (Czaki) [reply {{cookiecutter.plugin_name}}/.github/workflows/test_and_deploy.yml]: Yeah. They describe it as an easy migration, but we still stuck with this: https://github.com/napari/napari/pull/6701

**Review by Czaki — APPROVED** [review]

**Comment by Czaki** [comment]
Could you please update description of PR and remove `codecov` information?

**Pull request merged**